pkg/image/api/helper.go

```diff
@@ -157,12 +157,28 @@ func (r DockerImageReference) String() string {
 		registry += "/"
 	}
 
-	if len(r.Namespace) == 0 {
-		r.Namespace = DockerDefaultNamespace
+	namespace := r.Namespace
```

Comment:
Please undo this change - we explicitly do *not* manipulate these states during String, instead a caller should use `Minimal` or a new variant.